src/nft/components/bag/BagFooter.tsx

```diff
@@ -518,9 +518,27 @@ export const BagFooter = ({ totalEthPrice, fetchAssets, eventProperties }: BagFo
 
   const traceEventProperties = {
     usd_value: usdcValue?.toExact(),
+    using_erc20: !!inputCurrency,
     ...eventProperties,
   }
 
+  const tokenSelectedEventProperties = useMemo(() => {
+    if (!inputCurrency || inputCurrency.isNative) {
+      return undefined
+    }
+
+    return {
+      token_address: inputCurrency.address,
+      token_symbol: inputCurrency.symbol,
+    }
+  }, [inputCurrency])
+
+  useEffect(() => {
```

Comment:
Instead of this useEffect could we trigger this event manually via the `onCurrencySelect` function in the `CurrencySearchModal`?